projectq/ops/_basics.py

```diff
@@ -295,23 +295,29 @@ def get_merged(self, other):
 
     def __eq__(self, other):
         """ Return True if same class and same rotation angle. """
-        tolerance = 1.e-12
-        if isinstance(other, self.__class__):
-            difference = abs(self._angle - other._angle)
```

Comment:
Oh I see, you'd keep the double comparison.

The advantage of signed_mod is that it puts close differences near zero, so you don't have to remember to do two comparisons. It wraps you into [-c/2, c/2] instead of [0, c]. This is almost always what you want when doing comparisons, or when deciding "which direction should I turn to get there?" when working modulo a constant.